ext/date/date_core.c

```diff
@@ -3527,19 +3529,64 @@ date_initialize(int argc, VALUE *argv, VALUE self)
 
 /*
  * call-seq:
- *    Date.commercial([cwyear=-4712[, cweek=1[, cwday=1[, start=Date::ITALY]]]])  ->  date
+ *   Date.commercial(cwyear=-4712, cweek=1, cwday=1, start=Date::ITALY) -> date
+ *
+ * Returns a new \Date object constructed from the arguments.
+ *
+ * Argument +cwyear+ should be an integer.
+ * The actual year in the new date is:
+ *
+ * - +cwyear+, if January 1 of that year is a Friday, Saturday,
+ *   Sunday, or Monday
+ *   (that is, if the first Monday in calendar year +cwyear+
+ *   is on or before January 4):
+ *
+ *     [2010, 2000, 2006, 2001].each do |year|
+ *       wday = Date::ABBR_DAYNAMES[Date.new(year, 1, 1).wday]
+ *       cdate = Date.commercial(year)
+ *       p [year, wday, cdate.to_s]
+ *     end
+ *
+ *   Output:
+ *
+ *     [2010, "Fri", "2010-01-04"]
+ *     [2000, "Sat", "2000-01-03"]
+ *     [2006, "Sun", "2006-01-02"]
+ *     [2001, "Mon", "2001-01-01"]
+ *
+ * - <tt>cwyear - 1</tt>, otherwise:
+ *
+ *     [2002, 2003, 2004].each do |year|
+ *       wday = Date::ABBR_DAYNAMES[Date.new(year, 1, 1).wday]
+ *       cdate = Date.commercial(year)
+ *       p [year, wday, cdate.to_s]
+ *     end
  *
- * Creates a date object denoting the given week date.
+ *   Output:
  *
- * The week and the day of week should be a negative or a positive
- * number (as a relative week/day from the end of year/week when
- * negative).  They should not be zero.
+ *     [2002, "Tue", "2001-12-31"]
+ *     [2003, "Wed", "2002-12-30"]
+ *     [2004, "Thu", "2003-12-29"]
```

Comment:
This is a corner case that's only true for the first week (also similar situation for the last week of the year). We shouldn't start off talking about the corner case.